src/PointwiseFunctions/AnalyticData/NewtonianEuler/KhInstability.hpp

```diff
@@ -0,0 +1,245 @@
+#include "Evolution/Systems/NewtonianEuler/Sources/NoSource.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticData/AnalyticData.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace NewtonianEuler {
+namespace AnalyticData {
+
+/*!
+ * \brief Initial data to simulate the Kelvin-Helmholtz instability.
+ *
+ * For comparison purposes, this class implements the planar shear of
+ * \cite Schaal2015 The initial state consists of a horizontal strip of mass
+ * density \f$\rho_\text{in}\f$ moving with horizontal speed
+ * \f$v_{\text{in}}\f$. The rest of the fluid possesses mass density
+ * \f$\rho_\text{out}\f$, and its horizontal velocity is \f$v_{\text{out}}\f$,
+ * both constant. Mathematically,
+ *
+ * \f{align*}
+ * \rho(x, y) =
+ * \begin{cases}
+ * \rho_\text{in}, & \left|y - y_\text{mid}\right| < b/2\\
+ * \rho_\text{out}, & \text{otherwise},
+ * \end{cases}
+ * \f}
+ *
+ * and
+ *
+ * \f{align*}
+ * v_x(x, y) =
+ * \begin{cases}
+ * v_{\text{in}}, & \left|y - y_\text{mid}\right| < b/2\\
+ * v_{\text{out}}, & \text{otherwise},
+ * \end{cases}
+ * \f}
+ *
+ * where \f$b > 0\f$ is the thickness of the strip, and \f$y = y_\text{mid}\f$
+ * is its horizontal bimedian. The initial pressure is set equal to a constant,
+ * and the system is evolved assuming an ideal fluid of known adiabatic index.
+ * Finally, in order to excite the instability, the vertical velocity is
+ * initialized to a small perturbation
+ *
+ * \f{align*}
+ * v_y(x, y) = A\sin(4\pi x)
+ * \left[\exp\left(-\dfrac{(y - y_\text{top})^2}{2\sigma^2}\right) +
+ * \exp\left(-\dfrac{(y - y_\text{bot})^2}{2\sigma^2}\right)\right],
+ * \f}
+ *
+ * where \f$A\f$ is the amplitude, \f$\sigma\f$ is a damping factor, and
+ * \f$y_\text{top} = y_\text{mid} + b/2\f$ and
+ * \f$y_\text{bot} = y_\text{mid} - b/2\f$ are the vertical coordinates
+ * of the top and bottom boundaries of the strip, respectively.
+ *
+ * \note This class can be used to initialize 2D and 3D data. In 3D, the strip
+ * is aligned to the \f$x-\f$axis, and the vertical direction is taken to be
+ * the \f$z-\f$axis.
+ */
+template <size_t Dim>
+class KhInstability : public MarkAsAnalyticData {
+ public:
+  using equation_of_state_type = EquationsOfState::IdealFluid<false>;
+  using source_term_type = Sources::NoSource;
+
+  /// The adiabatic index of the fluid.
+  struct AdiabaticIndex {
+    using type = double;
+    static constexpr OptionString help = {"The adiabatic index of the fluid."};
+  };
+
+  /// The vertical coordinate of the horizontal bimedian of the strip.
+  struct StripBimedian {
```

Comment:
[opinion:] `StripBimedian` might be more mathematically precise, but I think using something like `StripCenter` or `StripHeight` or `StripPosition` would be easier to understand... both in the name of this struct and for the associated variable names and documentation.